packages/cli/src/CredentialsHelper.ts

```diff
@@ -462,8 +470,13 @@ export class CredentialsHelper extends ICredentialsHelper {
 		const credential = await this.sharedCredentialsRepository.findOne({
 			where: {
 				role: 'credential:owner',
-				user: {
-					role: 'global:owner',
+				project: {
+					projectRelations: {
+						role: 'project:personalOwner',
+						user: {
+							role: 'global:owner',
```

Comment:
We need a scope for accessing external secrets, right? Because `credential:read` would also allow members to access them.

Reply:
I hard coded the roles for now, but I'd like to have something like this here:

```ts
		const credentialRoles = this.roleService.rolesWithScope('credential', 'credential:read');
		const projectRoles = this.roleService.rolesWithScope('project', 'credential:read');
		const globalRoles = this.roleService.rolesWithScope('global', 'credential:read');

		const credential = await this.sharedCredentialsRepository.findOne({
			where: {
				role: In(credentialRoles),
				project: {
					projectRelations: {
						role: In(projectRoles),
						user: {
							role: In(globalRoles),
						},
					},
				},
				credentials: {
					id: nodeCredential.id,
				},
			},
		});
```

Reply:
Based on `packages/@n8n/permissions/src/types.ts` we do have a scope called `externalSecret:use`.

```
export type ExternalSecretScope = ResourceScope<'externalSecret', 'list' | 'use'>;
```

Maybe we could just use this.

Reply:
Thank, I'll check that out.

Reply:
Before the user needed to
1. own the credential (`credential:owner`)
2. be a global owner (`global:owner`)

The first one translates to the roles `project:personalOwner` and `credential:owner`.
The second one is now: `global:owner` or `global:admin`.

So in total
* `project:personalOwner`
* `credential:owner`
* `global:owner` | `global:admin`

Now the permissions necessary that I think make sense are:
* `credential:read` & `externalSecret:use`

But there is no credentials role and no project role with the `externalSecret:use` permission:

```ts
		const credentialRoles = this.roleService.rolesWithScope('credential', [
			'credential:read',
			'externalSecret:use',
		]); // = []
		const projectRoles = this.roleService.rolesWithScope('project', [
			'credential:read',
			'externalSecret:use',
		]); // = []
		const globalRoles = this.roleService.rolesWithScope('global', [
			'credential:read',
			'externalSecret:use',
		]); // = [ 'global:owner', 'global:admin' ]
```

I could also not require the `externalSecret:use` permission for the credential and project role, but that means then that the user does not need to own the credential, which make this behave differently than before. Also it feels wrong to only expect some permissions for some relationships.

```ts
		// [ 'credential:owner', 'credential:user' ]
		const credentialRoles = this.roleService.rolesWithScope('credential', ['credential:read']);
		// [ 'project:admin', 'project:personalOwner', 'project:editor', 'project:viewer' ]
		const projectRoles = this.roleService.rolesWithScope('project', ['credential:read']);
		[ 'global:owner', 'global:admin' ]
		const globalRoles = this.roleService.rolesWithScope('global', [
			'credential:read',
			'externalSecret:use',
		]);
```

This would behave like before, but it's messy:

```ts
		// [ 'credential:owner' ],
		const credentialRoles = this.roleService.rolesWithScope('credential', ['credential:update']);
		// [ 'project:admin', 'project:personalOwner', 'project:editor' ],
		const projectRoles = this.roleService.rolesWithScope('project', ['credential:update']);
		// [ 'global:owner', 'global:admin' ]
		const globalRoles = this.roleService.rolesWithScope('global', [
			'credential:read',
			'externalSecret:use',
		]);
```

So I'm wondering if we either allow global owners to use external secrets within credentials they don't own, or if we add `externalSecret:use` to `project:owner` and `credential:owner`?